![imagen](./img/ejercicios.png)

# INFORMACIÓN DEL PROYECTO

### TÍTULO: Mercado del alquiler de apartamento turísticos de cara a una inmersión inmobiliaria

### TEMA

Tema = **"La idea es realizar un análisis desde el punto de vista de una empresa que esté pensando en introducirse en el mercado de alquileres turísticos en Madrid."**

### HIPÓTESIS

####  **Hipotesis_1** = "Encontrar datos que demuestren cuál sería el mejor tipo de vivienda para alquilar, mejor ubicación... de cara a maximizar la rentabilidad de la inversión "


## OBTENCIÓN DE LOS DATOS

### DATASETS Y FUENTES ALTERNATIVAS DE DATOS

Incluye aquí una vista del dataset o datasets de los que partirás para poder evaluar tu hipótesis. <br>
También incluye el origen de estos datos y su fuente.

In [17]:
import pandas as pd
# guarda en variables los datasets y su fuente
df_1 = pd.read_csv("./data/calendar.csv.gz")
fuente_1 = "https://insideairbnb.com/get-the-data/" #El dataset que tengo yo lo encontré por internet, pero no lo descargué de esta página directamente. Pero quien se lo descargara fue de aquí.
df_2 = pd.read_csv("./data/idealista.csv")
fuente_2 = "https://www.idealista.com/sala-de-prensa/informes-precio-vivienda/venta/madrid-comunidad/report/" 
df_3 = pd.read_csv("./data/listings.csv")
fuente_3 = "https://insideairbnb.com/get-the-data/"
df_4 = pd.read_csv("./data/listings.csv.gz")
fuente_4 = "https://insideairbnb.com/get-the-data/"
df_5 = pd.read_csv("./data/neighbourhoods.csv")
fuente_5 = "https://insideairbnb.com/get-the-data/"
df_6 = pd.read_json("./data/neighbourhoods.geojson")
fuente_6 = "https://insideairbnb.com/get-the-data/"
df_7 = pd.read_csv("./data/reviews.csv")
fuente_7 = "https://insideairbnb.com/get-the-data/"
df_8 = pd.read_csv("./data/reviews.csv.gz")
fuente_ = "https://insideairbnb.com/get-the-data/"

Muestra mediante un head() los principales datasets con los que vas a trabajar

In [4]:
# Primer dataset. Calendario de fechas 
df_1.head(5)

listing_id        date available   price adjusted_price  minimum_nights  \
0        6369  2021-09-11         t  $60.00         $60.00             1.0   
1        6369  2021-09-12         t  $60.00         $60.00             1.0   
2        6369  2021-09-13         t  $60.00         $60.00             1.0   
3        6369  2021-09-14         t  $60.00         $60.00             1.0   
4        6369  2021-09-15         t  $60.00         $60.00             1.0   

   maximum_nights  
0          1125.0  
1          1125.0  
2          1125.0  
3          1125.0  
4          1125.0

In [5]:
# Segundo dataset. Listado del precios por m2 en los distintos distritos de Madrid
df_2.head(5)

table__cell        table__cell 2 table__cell 3 table__cell 4 table__cell 5  \
0      Madrid           3.683 €/m2       - 0,7 %       - 2,0 %       + 0,9 %   
1  4.085 €/m2  4.096 €/m2 jul 2019       - 0,3 %       + 4,5 %       + 0,4 %   
2  3.409 €/m2  3.663 €/m2 mar 2009       - 6,9 %       + 7,4 %       + 1,8 %   
3  2.123 €/m2                0,0 %       - 0,5 %       - 2,3 %           NaN   
4  4.827 €/m2  5.096 €/m2 ene 2020       - 1,7 %       - 5,3 %       + 0,5 %   

  table__cell 7   icon-elbow  \
0       - 3,6 %          NaN   
1       + 2,1 %   Arganzuela   
2       + 5,7 %      Barajas   
3      - 33,1 %  Carabanchel   
4       + 0,7 %       Centro   

                                     icon-elbow href  
0                                                NaN  
1  https://www.idealista.com/sala-de-prensa/infor...  
2  https://www.idealista.com/sala-de-prensa/infor...  
3  https://www.idealista.com/sala-de-prensa/infor...  
4  https://www.idealista.com/sala-de-prensa/infor...

In [9]:
# Tercer dataset. Listado de datos usuarios y viviendas reducido
df_3.head(5)

id                                      name    host_id host_name  \
0   6369  Rooftop terrace room ,  ensuite bathroom      13660     Simon   
1  21853                      Bright and airy room      83531     Abdel   
2  23001         Apartmento Arganzuela- Madrid Rio      82175     Jesus   
3  24805                    Gran Via Studio Madrid  346366726         A   
4  26825        Single Room whith private Bathroom     114340  Agustina   

  neighbourhood_group   neighbourhood  latitude  longitude        room_type  \
0           Chamartín  Hispanoamérica  40.45724   -3.67688     Private room   
1              Latina        Cármenes  40.40381   -3.74130     Private room   
2          Arganzuela         Legazpi  40.38840   -3.69511  Entire home/apt   
3              Centro     Universidad  40.42183   -3.70529  Entire home/apt   
4          Arganzuela         Legazpi  40.38975   -3.69018     Private room   

   price  minimum_nights  number_of_reviews last_review  reviews_per_month  \
0     60               1                 80  2019-05-14               1.21   
1     31               4                 33  2018-05-29               0.39   
2     50              15                  0         NaN                NaN   
3     92               5                 12  2020-03-01               0.51   
4     26               2                149  2018-05-15               1.65   

   calculated_host_listings_count  availability_365  number_of_reviews_ltm  \
0                               2               180                      4   
1                               2               364                      0   
2                               5               222                      0   
3                               1               115                      2   
4                               1               349                      0   

  license  
0     NaN  
1     NaN  
2     NaN  
3     NaN  
4     NaN

In [10]:
# Cuarto dataset. Listado de datos usuarios y viviendas ampliado
df_4.head(5)

id                         listing_url       scrape_id last_scraped  \
0   6369   https://www.airbnb.com/rooms/6369  20210910193531   2021-09-11   
1  21853  https://www.airbnb.com/rooms/21853  20210910193531   2021-09-11   
2  23001  https://www.airbnb.com/rooms/23001  20210910193531   2021-09-11   
3  24805  https://www.airbnb.com/rooms/24805  20210910193531   2021-09-11   
4  26825  https://www.airbnb.com/rooms/26825  20210910193531   2021-09-11   

                                       name  \
0  Rooftop terrace room ,  ensuite bathroom   
1                      Bright and airy room   
2         Apartmento Arganzuela- Madrid Rio   
3                    Gran Via Studio Madrid   
4        Single Room whith private Bathroom   

                                         description  \
0  Excellent connection with the AIRPORT and EXHI...   
1  We have a quiet and sunny room with a good vie...   
2  Apartamento de tres dormitorios dobles, gran s...   
3  Studio located 50 meters from Gran Via, next t...   
4  Nice and cozy roon for one person with a priva...   

                               neighborhood_overview  \
0                                                NaN   
1  We live in a leafy neighbourhood with plenty o...   
2  Barrio Arganzuela, junto a Madrid Rio, zonas c...   
3  The area is next to the Gran Via, so people li...   
4  Es un barrio muy tranquilo, en una zona de Mad...   

                                         picture_url    host_id  \
0  https://a0.muscache.com/pictures/683224/4cc318...      13660   
1  https://a0.muscache.com/pictures/68483181/87bc...      83531   
2  https://a0.muscache.com/pictures/58e6a770-509c...      82175   
3  https://a0.muscache.com/pictures/miso/Hosting-...  346366726   
4  https://a0.muscache.com/pictures/149358/218d5b...     114340   

                                      host_url  ...  \
0      https://www.airbnb.com/users/show/13660  ...   
1      https://www.airbnb.com/users/show/83531  ...   
2      https://www.airbnb.com/users/show/82175  ...   
3  https://www.airbnb.com/users/show/346366726  ...   
4     https://www.airbnb.com/users/show/114340  ...   

  review_scores_communication review_scores_location review_scores_value  \
0                        4.89                   4.77                4.85   
1                        4.82                   4.21                4.67   
2                         NaN                    NaN                 NaN   
3                        5.00                   5.00                4.83   
4                        4.71                   4.70                4.72   

  license instant_bookable calculated_host_listings_count  \
0     NaN                f                              2   
1     NaN                f                              2   
2     NaN                f                              5   
3     NaN                f                              1   
4     NaN                f                              1   

  calculated_host_listings_count_entire_homes  \
0                                           0   
1                                           0   
2                                           4   
3                                           1   
4                                           0   

  calculated_host_listings_count_private_rooms  \
0                                            2   
1                                            2   
2                                            1   
3                                            0   
4                                            1   

  calculated_host_listings_count_shared_rooms reviews_per_month  
0                                           0              1.21  
1                                           0              0.39  
2                                           0               NaN  
3                                           0              0.51  
4                                           0              1.65  

[5 rows x 74 columns]

In [11]:
# Quinto dataset. Barrios por distrito
df_5.head(5)

neighbourhood_group neighbourhood
0          Arganzuela       Acacias
1          Arganzuela        Atocha
2          Arganzuela       Chopera
3          Arganzuela      Delicias
4          Arganzuela      Imperial

In [18]:
# Sexto dataset. Barrios ampliado
df_6.head(5)

type                                           features
0  FeatureCollection  {'type': 'Feature', 'geometry': {'type': 'Mult...
1  FeatureCollection  {'type': 'Feature', 'geometry': {'type': 'Mult...
2  FeatureCollection  {'type': 'Feature', 'geometry': {'type': 'Mult...
3  FeatureCollection  {'type': 'Feature', 'geometry': {'type': 'Mult...
4  FeatureCollection  {'type': 'Feature', 'geometry': {'type': 'Mult...

In [19]:
# Séptimo dataset. Reviews
df_7.head(5)

listing_id        date
0        6369  2010-03-14
1        6369  2010-03-23
2        6369  2010-04-10
3        6369  2010-04-21
4        6369  2010-04-26

In [20]:
# Octavo dataset. Reviews ampliado
df_8.head(5)

listing_id     id        date  reviewer_id reviewer_name  \
0        6369  29428  2010-03-14        84790         Nancy   
1        6369  31018  2010-03-23        84338         David   
2        6369  34694  2010-04-10        98655        Marion   
3        6369  37146  2010-04-21       109871          Kurt   
4        6369  38168  2010-04-26        98901        Dennis   

                                            comments  
0  Simon and Arturo have the ultimate location in...  
1  Myself and Kristy originally planned on stayin...  
2  We had a great time at Arturo and Simon's ! A ...  
3  I very much enjoyed the stay.  \r<br/>It's a w...  
4  Arturo and Simon are polite and friendly hosts...

### Resumen de dataset y variables principales identificadas inicialmente
- Después de analizar los archivos entiendo que los principales son:
    - listings.csv (es una síntesis del ampliado)
    - listings.csv.gz (de donde puedo sacar algún campo más no incluido en el anterior)
    - idealista.csv (de donde puedo sacar datos para tratar de obtener lo que sería el importe de la compra de cada piso, si bien no vienen los m2 en ningún archivo.)
- Me han dado acceso a la api de idealista. Voy a ver si puedo completar por ahí.

#### Variables listado listings.csv

Tabla de variables listings.csv

|Nombre variable| Mide| Notas|
|---|---|--| 
|id| Identificador único| Se podría usar como índice|
|name| Es una lista de partes de la casa| Se podría utlizar las regex para sacar elementos comunes| 
|host_id| Número de huésped| Parece que se repite por lo que habrá usuarios que tengan más de una casa|
|host_name| Nombre del anfitrion| No creo que nos de mucha información|
|neighbourhood_group| Distrito en el que se encuentra la vivienda| *Interesante* para el análisis|
|neighbourhood| Barrio dentro del distrito| Interesante para análisis|
|latitude| la latido en una lista| No aporta valor|
|longitude| La loingitud| no aporta valor|
|room_type| Solo aparece somo habitación privada| no creo que sea muy relevante|
|price| precio por noche en euros| **importante** para el análisis|
|minimum_nights'| Número mínimo de noches| interesante para el análisis|
|number_of_reviews'| Nº de reseñas| interesante|
|last_review'| Fecha con la última reseña realizada| no aporta mucho valor y además tiene un 26% de nulos|
|reviews_per_month| proporción de reseñas al mes por usuario| puede ser interesante, pero tiene un 26% de nulos|
|calculated_host_listings_count'| Numero de propiedades dadas de alta por un usuario único| interesante|
|availability_365| Nº de días disponbles al año que está el alojamiento| **importante**|
|number_of_reviews_ltm| Número de reseñas en los últimos 12 meses| interesante|
|license| Si tiene o no licencia| No tiene caso información el 85% son nulos|


#### Variables listado listings_ampliado.csv

##### Análisis de variables
Tiene 74 variables, muchas de ellas tiene un número de nulos muy alto y otras no nos van a aportar información.
Hago la tabla de estas variables para las que me voy a quedar:

|Nombre variable| Mide| Notas| Importancia|
|---|---|---|---|
|id| referencia de cada inmueble| va a ser la conexión con la otra tabla| 0|
|description| texto breve sobre el inmueble| tanto en español como en inglés. puede ser interesante sacar algún tipo de relación entre la descripcion y los alquileres|1|
|amenities| describen los servicios que tiene el inmueble| idem que antes|1|
|bathrooms_text| número de baños y si es privado| valorar su peso de cara al alquiler|1|
|beds| número de camas| relevante de cara al alquiler?|1|
|accommodates| número de personas máx (entiendo) por inmueble alquilado| importante de cara al precio medio dado que el precio es por noche|0|
|bedrooms| numero de habitaciones del inmueble| valorar su peso de cara al alquiler|1|
|review_scores_rating| valoración media del inmueble por parte de los usuarios| valorar de cara a los alquileres|1|
|review_scores_communication| entiendo que la valoración de las comunicaciones por la situación del inmueble| interesante de cara a los alquileres|1|
|review_scores_location| valoración de la localización del inmueble|idem que antes|1|